ask-sdk-servlet-support/src/com/amazon/ask/servlet/ServletConstants.java

```diff
@@ -86,8 +90,29 @@ private ServletConstants() { }
     public static final long DEFAULT_TOLERANCE_MILLIS = 30000;
 
     /**
-     * Maximum allowed timestamp offset tolerance value in millis.
+     * Maximum allowed timestamp offset tolerance value in millis for standard skill requests.
+     */
+    public static final long MAXIMUM_TOLERANCE_MILLIS = 150000;
+
+    /**
+     * Timestamp offset tolerance value in millis for skill event requests. These requests may be
+     * delivered up to one hour past their original timestamp.
      */
-    public static final long MAXIMUM_TOLERANCE_MILLIS = 3600000;
+    public static final long TOLERANCE_SKILL_EVENTS_MILLIS = 3600000;
+
+    /**
+     * Set of skill event requests. Due to the delivery characteristics of events, these
+     * requests may arrive with a timestamp value of up to 1 hour prior to the current time
+     * and are verified against a different upper bound value than standard requests.
+     *
+     * https://developer.amazon.com/en-US/docs/alexa/smapi/skill-events-in-alexa-skills.html#delivery-of-events-to-the-skill
+     */
+    public static final Set<String> SKILL_EVENT_REQUESTS = Stream.of(
```

Comment:
Seems there's missing one called ProactiveSubscriptionChanged: https://github.com/alexa/alexa-apis-for-java/blob/59ed587fabe03ef257598bdfcf4c02e174a8d555/ask-sdk-model/src/com/amazon/ask/model/events/skillevents/ProactiveSubscriptionChangedRequest.java#L37 ?

By making this list, is it means whenever there's a new event, we have to update this list ?

Reply:
Maybe a bad idea: Instead of the exact string match, we could check whether the request type has the `AlexaSkillEvent` prefix

Reply:
Yeah, so originally I was going to go with the approach of whitelisting the entire namespace, but the `ProactiveSubscriptionChanged` request is not documented with the same behavior and there is no guarantee that any future requests would be so unfortunately I think we need to maintain an exact whitelist.

Reply:
Got it. Thanks for explanation 👍